crates/primitives/src/chain/spec.rs

```diff
@@ -361,14 +367,29 @@ pub static BASE_MAINNET: Lazy<Arc<ChainSpec>> = Lazy::new(|| {
             (Hardfork::Bedrock, ForkCondition::Block(0)),
             (Hardfork::Regolith, ForkCondition::Timestamp(0)),
         ]),
-        base_fee_params: BaseFeeParams::optimism(),
+        base_fee_params: BaseFeeParamsWrapper::Variable(vec![
+            (Hardfork::London, BaseFeeParams::optimism()),
+            (Hardfork::Canyon, BaseFeeParams::optimism_canyon()),
+        ]),
         prune_delete_limit: 1700,
         snapshot_block_interval: 1_000_000,
         ..Default::default()
     }
     .into()
 });
 
+/// A wrapper around [BaseFeeParams] that allows for specifying constant or dynamic EIP-1559
+/// parameters based on the active [Hardfork].
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum BaseFeeParamsWrapper {
+    /// Constant [BaseFeeParams]; used for chains that don't have dynamic EIP-1559 parameters
+    Constant(BaseFeeParams),
+    /// Variable [BaseFeeParams]; used for chains that have dynamic EIP-1559 parameters like
+    /// Optimism
+    Variable(Vec<(Hardfork, BaseFeeParams)>),
```

Comment:
I would like to have this as a standalone type like

`struct ForkBaseFeeBarams(Vec<..>)`